File: src/main/java/io/myosotisdev/utopianism/Locales.java
package io.myosotisdev.utopianism;

import net.minestom.server.utils.NamespaceID;

import java.util.HashMap;
import java.util.Locale;

public class Locales
{
    public static Locale DefaultLocaleSetting = Locale.ENGLISH;
    private static final String EMPTY = "NO-MATCHED-STRING";
    private static HashMap<Locale, HashMap<NamespaceID, String>> Locales = new HashMap<>();

    public static String get(Locale locale, NamespaceID namespace)
    {
        HashMap<NamespaceID, String> localeStrings = Locales.get(locale);
        if(localeStrings != null)
            return localeStrings.getOrDefault(namespace, EMPTY);
        else
            return Locales.get(DefaultLocaleSetting).getOrDefault(namespace, EMPTY);
    }

    public static String get(Locale locale, String namespace)
    {
        return get(locale, NamespaceID.from("utopianism", namespace));
    }
}

File: src/main/java/io/myosotisdev/utopianism/Locales.kt
package io.myosotisdev.utopianism

import io.myosotisdev.minestom.Minestom
import io.myosotisdev.minestom.module.LocaleModule
import net.minestom.server.entity.Player
import net.minestom.server.utils.NamespaceID
import java.util.*

object Locales
{
    private val localeModule: LocaleModule? = Minestom.getModule("locale") as LocaleModule?
    fun translate(locale: Locale?, namespace: NamespaceID): String
    {
        return localeModule!!.translate(locale, namespace)
    }

    fun translate(locale: Locale?, string: String): String
    {
        return translate(locale, NamespaceID.from(string!!))
    }

    fun translate(player: Player?, namespace: NamespaceID): String
    {
        return translate(player?.locale ?: null, namespace)
    }

    fun translate(player: Player?, string: String): String
    {
        return translate(player?.locale ?: null, NamespaceID.from(string!!))
    }

    fun translate(string: String?): String
    {
        return translate(localeModule!!.defaultLocale, NamespaceID.from(string!!))
    }
}